src/include/duckdb.h

```diff
@@ -4485,6 +4522,82 @@ DUCKDB_C_API char *duckdb_table_description_get_column_name(duckdb_table_descrip
 // Arrow Interface
 //===--------------------------------------------------------------------===//
 
+/*!
+Transforms a DuckDB Schema into an Arrow Schema
+
+* @param client_properties The client properties to extract the arrow settings from.
+* @param types The DuckDB Logical Types for each column in the schema.
+* @param names The names for each column in the schema.
+* @param column_count The number of columns that exist in the schema.
+* @param out_schema The resulting arrow schema. Must be destroyed with `duckdb_destroy_arrow_schema`.
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data duckdb_to_arrow_schema(duckdb_client_properties *client_properties,
+                                                      duckdb_logical_type *types, char **names, idx_t column_count,
+                                                      duckdb_arrow_schema *out_schema);
+
+/*!
+Transforms a DuckDB data chunk into an Arrow array.
+
+* @param client_properties The client properties to extract the Arrow settings from.
+* @param chunk The DuckDB data chunk to convert.
+* @param out_arrow_array The output Arrow structure that will hold the converted data. Must be destroyed with
+`duckdb_arrow_array`
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data duckdb_data_chunk_to_arrow(duckdb_client_properties *client_properties,
+                                                          duckdb_data_chunk chunk, duckdb_arrow_array *out_arrow_array);
+
+/*!
+Transforms an Arrow Schema into a DuckDB Schema.
+
+* @param connection The connection to get the transformation settings from.
+* @param schema The input Arrow schema.
```

Comment:
We should mention here that ownership is passed (i.e the caller shouldn't call `release()` on it anymore)

Reply:
I think that is a fair criticism. I'll have a go without the wrapper, that seems to make most sense to me too.